reconcile/aws_version_sync/integration.py

```diff
@@ -170,8 +176,34 @@ def get_aws_metrics(
                         resource_engine_version=m["engine_version"],
                     )
                     for m in prom_get_func(
-                        cluster.prometheus_url,
-                        {"query": "aws_resources_exporter_rds_engineversion"},
+                        url=cluster.prometheus_url,
+                        params={"query": "aws_resources_exporter_rds_engineversion"},
+                        token=token,
+                        timeout=timeout,
+                    )
+                ]
+                # ElastiCache resources
+                metrics += [
+                    ExternalResource(
+                        provider="aws",
+                        provisioner=ExternalResourceProvisioner(
+                            uid=m["aws_account_id"]
+                        ),
+                        resource_provider="elasticache",
+                        # cache_name is the ElastiCache node name, which derived from replication_group_id not resource_identifier!
+                        resource_identifier=elasticache_replication_group_id_to_identifier.get(
+                            node_name_to_cache_name(m["cache_name"]),
+                            node_name_to_cache_name(m["cache_name"]),
+                        ),
+                        resource_engine=m["engine"],
+                        resource_engine_version=m["engine_version"],
+                    )
+                    for m in prom_get_func(
+                        url=cluster.prometheus_url,
+                        params={
+                            # use the first Redis node of the cluster to get the version
```

Comment:
What about using an aggregation function like max? node with 1$ is always there?

Reply:
See https://github.com/app-sre/qontract-reconcile/pull/3986#discussion_r1444291899